### PR TITLE
MAINT : Refactoring lil List of Lists

### PR DESCRIPTION
**Pull request by AnirudhDagar**
This PR fixes the lil format from LInked List to List of Lists.

### CODE REVIEW

**Review by tylerjereddy — APPROVED** [review]
This appears to based on the suggestion from Ralf in https://github.com/scipy/scipy/pull/10847#issuecomment-533781022

So probably sensible, though not my area of expertise

File: scipy/sparse/base.py [hunk 1]
```diff
@@ -910,7 +910,7 @@ def tocoo(self, copy=False):
         return self.tocsr(copy=False).tocoo(copy=copy)
 
     def tolil(self, copy=False):
-        """Convert this matrix to LInked List format.
+        """Convert this matrix to List of List format.
```
Comment (perimosocordiae) [inline scipy/sparse/base.py]: List of Lists

**Review by perimosocordiae — APPROVED** [review]
Seems unlikely to break backwards compatibility, so I'm +1 to merge.

File: scipy/sparse/base.py (hunk 1 above)
Comment (AnirudhDagar) [reply scipy/sparse/base.py]: Done!

**Comment by tylerjereddy** [comment]
Great, merging, thanks

**Pull request merged**